### PR TITLE
feat: Added Script Window

### PR DESCRIPTION
**Pull request by GustavoLR548**
# Script Window for generated GDScript code

Instead of just printing the results to the screen, I created a Window to popup with the generated script. 

You can also copy the entire generated code with a press of a button

If there is any problem with the code or anything you would like to change, please let me know!

Edit: The "Copiar" is misleading, I inserted the text as "Copy", but when running the code, it translated to "Copiar" for some reason that I cannot explain.

image

### CODE REVIEW

File: addons/block_code/plugin.cfg
Comment (dbnicholson) [inline addons/block_code/plugin.cfg:6]: Can you leave this out of the patch? It's there for when we make releases.

File: addons/block_code/ui/main_panel.gd
Comment (dbnicholson) [inline addons/block_code/ui/main_panel.gd:69]: Can you remove the commented lines?

File: addons/block_code/ui/main_panel.gd
Comment (dbnicholson) [inline addons/block_code/ui/main_panel.gd:4]: Can we call this signal `script_window_requested`?

File: addons/block_code/ui/script_window/script_window.gd
Comment (dbnicholson) [inline addons/block_code/ui/script_window/script_window.gd:15]: I don't think this is ever connected, so I think it can be removed.

**Review by dbnicholson — CHANGES_REQUESTED** [review]
I like this and had been thinking about doing the same thing for a while. I left a few cleanup comments, but overall this looks good to me. Thanks!

File: addons/block_code/ui/script_window/script_window.tscn
Comment (dbnicholson) [inline addons/block_code/ui/script_window/script_window.tscn:31]: I think this would be better as the window title instead of a label in the window. I think I'd probably make it something a little more generic like `Block Code Generated GDScript`. But we can always fine tune it after merging.

**Comment by GustavoLR548** [comment]
Happy to see you liked the feature :).

Sure thing! I will make the requested changes.

**Comment by GustavoLR548** [comment]
If there is any other problem with the pull request, let me know ;-)

File: addons/block_code/ui/script_window/script_window.gd
Comment (manuq) [inline addons/block_code/ui/script_window/script_window.gd:3]: @GustavoLR548 please remove the class name. You are already preloading this script fine in `addons/block_code/block_code_plugin.gd` using the resource path. The problem we found with `class_name` was that they can collide with the user project. Suppose a user has a `ScriptWindow` class in their project and then they add this plugin, the names collide.
```suggestion
```

**Review by manuq — CHANGES_REQUESTED** [review]
Thank you for the contribution @GustavoLR548 ! One more detail.

**Comment by GustavoLR548** [comment]
Okay, I removed the class name.

Additionally, I changed the name of the node and the function called by the signal responsible for showing the code to reflect the changes made.

Let me know if you need any further adjustments.

File: addons/block_code/block_code_plugin.gd
Comment (manuq) [inline addons/block_code/block_code_plugin.gd:67]: @GustavoLR548 I noticed in your screenshot that there is no indentation. I think you can get indentation by replacing the tabs with 4 spaces:
```suggestion
	script_window.script_content = script.replace('\t', '    ')
```

And then add a monospace font to the label:
- Add Label Settings
- Add Font -> SystemFont
- Font Names -> Add Element -> Monospace
Captura desde 2024-08-01 13-55-34

The result will be something like this:
Captura desde 2024-08-01 14-00-30

**Review by manuq — CHANGES_REQUESTED** [review]

**Comment by GustavoLR548** [comment]
Here, not only I changed the font, but also discovered that you can add a syntax highlighter to the text, and so I decided to add :D

image

**Comment by GustavoLR548** [comment]
Ok, added the Monospace font. Also there is now a syntax highlighter :D

image

**Review by manuq — APPROVED** [review]
@GustavoLR548 with syntax highlight this is just perfect! Please squash your hotfixes into a single commit so we can merge.

**Comment by GustavoLR548** [comment]
Perfect! Just to inform I send another commit to add color to more keywords that were not present before.

image

Ok, I will squash the hotfixes

**Comment by manuq** [comment]
> Ok, I think I was able to correctly squash the commits into a single one.
> 
> If I made any mistakes or if there is any other adjustments to be made, let me know

Obrigado! This is now in a single commit. As a last request, can I ask you to change the commit description to describe the change? Something like the following would be enough:

> Show generated script in a window
>
> Replace printing the generated script to the output. Display it in a separate window instead.

**Comment by GustavoLR548** [comment]
De nada :)

Sure! I will change the commit description.

**Comment by manuq** [comment]
@GustavoLR548 your contribution is now merged and will be in next release, thank you! We have some checks and they were failing so I decided to fix them myself in #183 .

**Comment by dbnicholson** [comment]
In the future it might be best to have the contributor [allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) and update their branch directly. Then you won't need to open a separate PR on a separate branch.

**Comment by manuq** [comment]
> In the future it might be best to have the contributor [allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) and update their branch directly. Then you won't need to open a separate PR on a separate branch.

Oh I didn't know about that feature. Thanks @dbnicholson, I will do so next time.

**Pull request closed without merging**
